lib/services/nativescript-project-plugins-service.ts

```diff
@@ -0,0 +1,463 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import path = require("path");
+import * as util from "util";
+import {EOL} from "os";
+import Future = require("fibers/future");
+import * as shelljs from "shelljs";
+import * as semver from "semver";
+
+import PluginsDataLib = require("../plugins-data");
+
+export class NativeScriptProjectPluginsService implements IPluginsService {
+	private static MESSAGES = ["NPM Packages", "NPM NativeScript Plugins", "Marketplace Plugins"];
+
+	constructor(
+		private $errors: IErrors,
+		private $logger: ILogger,
+		private $project: Project.IProject,
+		private $projectConstants: Project.IProjectConstants,
+		private $fs: IFileSystem,
+		private $server: Server.IServer,
+		private $httpClient: Server.IHttpClient,
+		private $childProcess: IChildProcess) {
+		}
+
+	public getAvailablePlugins(pluginsCount?: number): IPlugin[] {
+		let count = pluginsCount || 10;
+		let futures = [
+			this.getUniqueMarketplacePlugins(),
+			this.getTopNpmPackages(count),
+			this.getTopNativeScriptNpmPackages(count)
+		];
+		Future.wait(futures);
```

Comment:
if one of the futures throws exception, the user will not receive any results. Is this expected? For example if npm is down, we will not show information for marketplace plugins as well.